Diagonal.java

```diff
@@ -0,0 +1,66 @@
+// Time Complexity : O(N * M) N = rows, M = columns
+// Space Complexity : O(1)
+// Did this code successfully run on Leetcode : Yes
+// Three line explanation of solution in plain english
```

Comment:
This is still sort of pseudocode, use sentences that give an overview and are not implementation details. 

An example would be-
-keep a track of three pointers representing row, column and direction. for every
-direction compute three cases which can occur and manipulate the pointers accordingly. 
-repeat the same until the pointers haven't reached the length and width of the matrix